IntelliTect.Multitool/Security/ClaimsPrincipalExtensions.cs

```diff
@@ -3,42 +3,46 @@
 using System.Linq;
 using System.Security.Claims;
 
-namespace IntelliTect.Multitool.Security
+namespace IntelliTect.Multitool.Security;
+
+/// <summary>
+/// Gets information from a <see cref="ClaimsPrincipal"/>
+/// </summary>
+public static class ClaimsPrincipalExtensions
 {
     /// <summary>
-    /// Gets information from a <see cref="ClaimsPrincipal"/>
+    /// Gets the user ID from a <see cref="ClaimsPrincipal"/>.
     /// </summary>
-    public static class ClaimsPrincipalExtensions
+    /// <param name="principal">The <see cref="ClaimsPrincipal"/> to find a user ID for.</param>
+    /// <returns>A <see cref="string"/>, or null if the <see cref="ClaimsPrincipal"/> doesn't contain a <see cref="ClaimTypes.NameIdentifier"/>.</returns>
+    /// <exception cref="ArgumentNullException">principal is null.</exception>
+    public static string? GetUserId(this ClaimsPrincipal principal)
     {
-        /// <summary>
-        /// Gets the user ID from a <see cref="ClaimsPrincipal"/>.
-        /// </summary>
-        /// <param name="principal">The <see cref="ClaimsPrincipal"/> to find a user ID for.</param>
-        /// <returns>A <see cref="string"/>, or null if the <see cref="ClaimsPrincipal"/> doesn't contain a <see cref="ClaimTypes.NameIdentifier"/>.</returns>
-        /// <exception cref="ArgumentNullException">principal is null.</exception>
-        public static string GetUserId(this ClaimsPrincipal principal)
-        {
-            if (principal == null) throw new ArgumentNullException(nameof(principal));
+        if (principal is null) throw new ArgumentNullException(nameof(principal));
 
-            Claim claim = principal.FindFirst(ClaimTypes.NameIdentifier);
-            if (claim != null) return claim.Value;
+        Claim? claim = principal.FindFirst(ClaimTypes.NameIdentifier);
+        if (claim is null)
+        {
             claim = principal.FindFirst(ClaimTypes.Name);
-            return claim?.Value;
+            string claimValue = claim is null ? throw new InvalidOperationException(nameof(claim)) : claim.Value;
+            return claimValue;
```

Comment:
Just to avoid changing behavior on consumers, I would keep the original behavior of silently ignoring the null claim
```suggestion
            return claim?.Value;
```